src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1659,51 +1659,87 @@ protected override void InvokeDefaultAction(string path)
                             return;
                         }
 
-                        bool attributeFilter = true;
-                        bool switchAttributeFilter = true;
-                        bool filterHidden = false;           // "Hidden" is specified somewhere in the expression
-                        bool switchFilterHidden = false;     // "Hidden" is specified somewhere in the parameters
-
-                        if (null != evaluator)
+                        // Internal test code, run only if one of the
+                        // 'GciEnumerationAction' test hooks are set.
+                        if (InternalTestHooks.GciEnumerationActionDelete)
                         {
-                            attributeFilter = evaluator.Evaluate(filesystemInfo.Attributes);  // expressions
-                            filterHidden = evaluator.ExistsInExpression(FileAttributes.Hidden);
+                            if (filesystemInfo.Name == "c283d143-2116-4809-bf11-4f7d61613f92")
+                            {
+                                var fullName = Path.Combine(directory.FullName, filesystemInfo.Name);
+                                File.Delete(fullName);
```

Comment:
I changed it to `filesystemInfo.FullName` and the test passed on windows. I didn't try on Linux or OSX. Are you seeing issue with `filesystemInfo.FullName`?